compiler/rustc_incremental/src/persist/save.rs

```diff
@@ -73,6 +61,20 @@ pub fn save_dep_graph(tcx: TyCtxt<'_>) {
                     }
                 });
             },
+            move || {
+                // We execute this after `incr_comp_persist_dep_graph` for the serial compiler
+                // to catch any potential query execution writing to the dep graph.
+                sess.time("incr_comp_persist_result_cache", || {
+                    // Drop the memory map so that we can remove the file and write to it.
+                    if let Some(odc) = &tcx.query_system.on_disk_cache {
+                        odc.drop_serialized_data(tcx);
+                    }
+
+                    file_format::save_in(sess, query_cache_path, "query cache", |e| {
+                        encode_query_cache(tcx, e)
+                    });
+                });
+            },
```

Comment:
Why are blocks swapped?
Should this block be wrapped inside a `dep_graph.with_query_deserialization` to ICE if a query is invoked?

Reply:
`typeck` and `mir_borrowck` still use queries in their `cache_on_disk_if` block, so that cannot be used yet. This block order can cause panics due to `GraphEncoder` being stolen though, so it will catch query execution at least.